std/traits.d

```diff
@@ -5971,6 +5972,31 @@ enum bool isNarrowString(T) = (is(T : const char[]) || is(T : const wchar[])) &&
     }
 }
 
+/**
+ * Detects whether `T` is a comparable type. Basic types and structs and
+ * classes that implement opCmp are comparable.
+ */
+enum bool isComparable(T) = is(typeof({ return T.init < T.init; }));
```

Comment:
```D
enum bool isComparable(T) = is(typeof(T.init < T.init) : bool);
```

Also I'm wondering if `isOrdered` or `supportsOrdering` would be a better name, since there's also the equality comparison. Probably if we add `isEquatable` there would be no mistake.